dali/pipeline/operators/reader/coco_reader_op_test.cc

```diff
@@ -0,0 +1,301 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include "dali/test/dali_test_config.h"
+#include "dali/pipeline/pipeline.h"
+
+namespace dali {
+
+class CocoReaderTest : public ::testing::Test {
+ public:
+  std::vector<std::pair<std::string, std::string>> Outputs() {
+    return {{"images", "cpu"}, {"boxes", "cpu"}, {"labels", "cpu"}, {"image_ids", "cpu"}};
+  }
+
+  OpSpec CocoReaderOpSpec() {
+    return OpSpec("COCOReader")
+          .AddArg("device", "cpu")
+          .AddArg("file_root", file_root_)
+          .AddArg("annotations_file", annotations_filename_)
+          .AddArg("save_img_ids", true)
+          .AddOutput("images", "cpu")
+          .AddOutput("boxes", "cpu")
+          .AddOutput("labels", "cpu")
+          .AddOutput("image_ids", "cpu");
+  }
+
+  int SmallCocoSize() { return 64; }
+  int EmptyImages() { return 2; }
+  int ImagesWithBigObjects() { return 2; }
+  int ObjectCount() { return 194; }
```

Comment:
How about making these functions `constexpr` (and free)?

Reply:
I don't know about it. It's not like they have any meaning without context of this class. They are directly connected with `CocoReaderTest` fields. And you can't make them `constexpr` as members, so I think I will leave this as is.